pkg/apis/awsprovider/v1alpha1/types.go

```diff
@@ -340,6 +335,40 @@ func (i IngressRules) Difference(o IngressRules) (out IngressRules) {
 	return
 }
 
+// Equal returns true if two IngressRule are equal
+func (i *IngressRule) Equal(o *IngressRule) bool {
```

Comment:
Can you please add some unit tests for this?

Reply:
Sure, I'll add.

Reply:
@ashish-amarnath I have added tests. PTAL.

Reply:
@vjdhama  Thank you for adding the unit tests :-)